tests/transforms/test_optimization/test_pattern_matching.py

```diff
@@ -0,0 +1,306 @@
+# Copyright 2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pennylane as qml
+from pennylane.transforms.optimization.pattern_matching import pattern_matching
+
+
+class TestPatternMatching:
```

Comment:
Do you refer to PennyLane templates or patter templates? For the second it is built for it. For PennyLane template it is not that simple, because we use the commutation DAG to apply the algorithm and this DAG uses truth table for commutation and some PennyLane templates did not feat into that. But you can use `pattern_matching`function for some of them. The circuit optimization `pattern_matching_optimization` is only available for templates that implement the identity.